mistralrs-core/src/pipeline/macros.rs

```diff
@@ -2,14 +2,85 @@
-            })
-            .unwrap_or_else(|e| {
+        if std::path::Path::new($model_id).is_absolute() {
+            let listing = std::fs::read_dir($model_id);
+            if listing.is_err() {
+                panic!("Cannot list directory {:?}", $model_id)
+            }
+            let listing = listing.unwrap();
+            listing
+                .into_iter()
+                .map(|s| {
+                    s.unwrap()
+                        .path()
+                        .file_name()
+                        .unwrap() // Should never terminate in `..`
+                        .to_str()
+                        .expect("Could not convert to str")
+                        .to_string()
+                })
+                .collect::<Vec<String>>()
+                .into_iter()
+        } else {
+            $api.info()
+                .map(|repo| {
+                    repo.siblings
+                        .iter()
+                        .map(|x| x.rfilename.clone())
+                        .collect::<Vec<String>>()
+                })
+                .unwrap_or_else(|e| {
+                    // If we do not get a 404, it was something else.
+                    let format = format!("{e:?}");
+                    let mut unauth = false;
+                    if let hf_hub::api::sync::ApiError::RequestError(resp) = e {
+                        let resp = resp.into_response();
+                        // If it's 401, assume that we're running locally only.
+                        if resp.as_ref().is_some_and(|r| r.status() == 401) {
+                            unauth = true;
+                        } else if resp.as_ref().is_some_and(|r| r.status() != 404) {
+                            panic!("{format}");
+                        }
+                    }
+
+                    let listing = std::fs::read_dir($model_id);
+                    if listing.is_err() && unauth {
+                        panic!("{format}");
+                    } else if listing.is_err() {
+                        panic!("Cannot list directory {:?}", $model_id)
+                    }
+                    let listing = listing.unwrap();
+                    listing
+                        .into_iter()
+                        .map(|s| {
+                            s.unwrap()
+                                .path()
+                                .file_name()
+                                .unwrap() // Should never terminate in `..`
+                                .to_str()
+                                .expect("Could not convert to str")
+                                .to_string()
+                        })
+                        .collect::<Vec<String>>()
+                })
+                .into_iter()
+        }
+    };
+}
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! api_get_file {
+    ($api:expr, $file:expr, $model_id:expr) => {
+        if std::path::Path::new($model_id).is_absolute() {
```

Comment:
Same here, if we could check if the file exists that may be more flexible.

Reply:
same above